docs/source/architecture/journal.rst

```diff
@@ -0,0 +1,375 @@
+consensus algorithms.  The Journal implements a consensus interface that
+defines the entry points and responsibilities of a consensus algorithm.
+
+The BlockStore
+==============
+
+The BlockStore contains all the blocks in the current block chain, that is the
+list of blocks from the current chain head back to the Genesis blocks. Blocks
+from forks are not included in the BlockStore. The BlockStore also includes a
+reference to the head of the current chain. The BlockStore is expected to be
+coherent at all times and an error in the BlockStore is considered a 
+non-recoverable error for the validator. Such critical errors would include
+missing blocks, bad indexes, missing chain reference, incomplete blocks or
+invalid blocks in the store. The BlockStore provides an atomic means to update
+the store when the current fork is changed (the chain head is updated).
+
+The BlockStore is a persistent on-disk store of all Blocks in the current
+chain. When the validator is started the contents of the BlockStore is trusted
+to be the current state of the world. The BlockStore also stores the identifier
+of the chain head.
+
+All blocks stored here are formally complete and include the block weight
+calculated during block validation. The BlockStore allows blocks to be accessed
+via Block ID, additionally blocks may be accessed via Batch ID or Transaction
+ID.  I.e. GetBlockByTransactionId, GetBatchByTransactionId and
+GetBlockByBatchId.
+
+The BlockStore maintains internal mappings of Transaction-to-Block and
+Batch-to-Block. These may be rebuilt if missing or corrupt, this rebuild should
+be done during start up, not during the course of normal operation. These
+mappings should be stored in a format that is cached to disk, so they are not
+required to be held in memory at all times. As the block-chain grows these will
+become quite large.
+
+The BlockStore provides an atomic method for updating the current head of the
+chain. In order for the BlockStore to switch forks it is provided with a list
+of blocks in the new chain to commit, and a list of blocks in the old chain to
+decommit. These lists are the blocks in each fork back to the common root.
+
+The BlockCache
+==============
+
+The Block Cache holds the working set of blocks for the validator and tracks the
+processing state. This processing state is tracked as valid, invalid, or
+unknown. Valid blocks are blocks that have been proven to be valid by the
+ChainController. Invalid blocks are blocks that failed validation or have an
+invalid block as a predecessor. Unknown are blocks that have not yet completed
+validation, usually having just arrived from the Completer.
+
+The BlockCache is an in memory construct, not persisted. It is rebuilt by demand
+when the system is started. 
+
+If a block is not present in the BlockCache it will look in the BlockStore for
+the block. If it not found the lookup fails, the block is unknown to the system.
```

Comment:
Add a comma after "not found" or maybe 'or'. This sentence doesn't parse currently.